### ISSUE TITLE
Using HTTP challenge provider results in pod stuck at ContainerCreating stage

### ISSUE THREAD

**Issue by tamalsaha** [appscode/voyager#455]
From slack: 
" Trying to create http ingress with certificate. Following this guide: https://github.com/appscode/voyager/blob/master/docs/user-guide/certificate/create.md#create-certificate-with-http-provider. After created the ingress, pod stuck at ContainerCreating stage because the xxxxxx-cert-secret-volume is not mounted. According to the guide, it suppose to be created after the cert resource is created. Looks like a chicken and egg problem. Did I miss anything? "

**Comment by xcompass** [comment]
I'm using bare metal and kind of forced to use HTTP. And using k8s 1.7.3, voyager 3.2.0-rc1. Let me know if you need more info.

**Comment by xcompass** [comment]
Here is the ingress and cert resource:
```yaml
apiVersion: voyager.appscode.com/v1beta1
kind: Ingress
metadata:
  name: internal-services-ingress-controller
  namespace: default
  annotations:
    kubernetes.io/ingress.class: internal
    ingress.appscode.com/type: HostPort
    ingress.appscode.com/node-selector: '{ "role": "internallb" }'
    ingress.appscode.com/stats: 'true'
spec:
  tls:
    - secretName: mydomain-com-cert
      hosts:
      - mydomain.com
  rules:
  - host: mydomain.com
    http:
      paths:
      - backend:
          serviceName: mydomain-service
          servicePort: '8200'
```

```yaml
apiVersion: voyager.appscode.com/v1beta1
kind: Certificate
metadata:
  name: mydomain-com-cert
  namespace: default
spec:
  domains:
  - mydomain.com
  email: myemail@mydomain.com
  provider: http
  httpProviderIngressReference:
    apiVersion: voyager.appscode.com/v1beta1
    kind: Ingress
    Namespace: default
    Name: internal-services-ingress-controller
```

**Comment by tamalsaha** [comment]
Hi, what DNS registrar are you using?

**Comment by xcompass** [comment]
I manage my own dns using a solution from BlueCat Networks. I manually created the domain name there.

**Comment by tamalsaha** [comment]
This is also addressed in #505

**Comment by tamalsaha** [comment]
Fixed in #506
